File: wallet.js
import prompt from 'prompt-sync';
import { show } from './function/show.js';
import { showBalance } from './function/showBalance.js';
import { addBalance } from './function/addBalance.js';
import { substractBalance } from './function/substractBalance.js';
import { showHistory } from './function/showHistory.js';
import { addCard } from './function/addCard.js';
import { getCard } from './function/getCard.js';
import { addPhoto } from './function/addPhoto.js';
import { getPhoto } from './function/getPhoto.js';

const wallet = () => {
    let balance = 5000;
    const promptFunction = prompt();
    let choice;
    let history = [];
    let cards = [];
    let photos = [];
    let wallet = {
        "owner": "Liana Finaritra"
    };
    while (true) {
        show('Appuyer sur un chiffre pour éxecuter une action: ' + 
        '\n 1 - Afficher le solde' + 
        '\n 2 - Déposer ' + 
        '\n 3 - Retirer' +
        '\n 4 - Historiques des transactions' +
        '\n 5 - Ajouter une carte' +
        '\n 6 - Récupérer une carte' +
        '\n 7 - Afficher les informations sur le portefeuille' +
        '\n 8 - Ajouter une photo' +
        '\n 9 - Récuperer une photo' +
        '\n 0 - Sortir \n');

        choice = promptFunction('Votre choix : ');
        
        switch (choice) {
            case '1':
                showBalance(balance);
                break
            case '2':
                const added = promptFunction('Montant à déposer (Ar) :');
                balance = addBalance(history, balance, added);
                break
            case '3':
                const substracted = promptFunction('Montant à retirer (Ar) :');
                balance = substractBalance(history, balance, substracted);
                break
            case '4':
                showHistory(history);
                break
            case '5':
                console.clear();
                const name = promptFunction('Le nom de la carte (CIN, CB, ...):');
                const owner = promptFunction('Son propriétaire: ')
                addCard(cards, name, owner);
                break
            case '6':   
                console.clear();
                const search = promptFunction('Quel carte voulez vous prendre :');
                cards = getCard(cards, search);
                break
            case '7':
                console.clear();
                show(`\n Ce portefeuille appartient à: ${wallet.owner} \n`);
                break
            case '8':
                console.clear();
                const photoID = promptFunction("Un identifiant pour la photo: ");
                const photoOwner = promptFunction('Son propriétaire: ');
                addPhoto(photos, photoID, photoOwner);
                break
            case '9':
                console.clear();
                const searchPhoto = promptFunction('Quel photo voulez vous prendre :');
                photos = getPhoto(photos, searchPhoto);
                break
            case '0':
                break;
            default:
                console.clear();
                show('\n Choix non valide. Veuillez saisir un chiffre de 1 à 4. \n');
                break;
            }
        if (choice === '0') {
            break;
        }
    }
};

wallet();

File: wallet.py
from clear import clear_console

class Wallet:
    def __init__(self):
        self.balance = 5000
        self.history = []
        self.cards = []
        self.photos = []
        self.wallet = {
            "owner": "Liana Finaritra"
        }

    def show_infos(self):
        print(f"\n Ce portefeuille appartient à: {self.wallet['owner']} \n")

    def add_transaction_to_history(self, type, value):
        new_transaction = {
            "type": type,
            "value": value
        }
        self.history.append(new_transaction)

    def show_history(self):
        clear_console()
        print(f"\n Historique: \n")
        for index, transaction in enumerate(self.history, start=1):
            print(f"{index}, Type: {transaction['type']}, Valeur: {transaction['value']} Ar \n")

    def show_balance(self):
        clear_console()
        print(f"Votre solde actuel est: {self.balance} Ar \n")

    def add_balance(self, money_to_add):
        add_value = int(money_to_add)
        self.balance = self.balance + add_value
        self.add_transaction_to_history("Dépot", add_value)
        clear_console()
        print(f"Votre nouveau solde est : {self.balance} Ar \n")

    def subtract_balance(self, money_to_subtract):
        subtract_value = int(money_to_subtract)
        if(subtract_value > self.balance):
            print("Vous n'avez pas assez d'argent pour ce retrait \n")

        else:
            self.balance = self.balance - subtract_value
            self.add_transaction_to_history("Retrait", subtract_value)
            clear_console()
            print(f"Votre nouveau solde est : {self.balance} Ar \n")

    def add_card(self, name, owner):
        new_card = {
            "name": name,
            "owner": owner
        }
        self.cards.append(new_card)
        print(f'\n Ajout réussi \n')

    def get_card(self, name):
        search_card = next((item for item in self.cards if item['name'] == name), None)
        updatedCards = [item for item in self.cards if item['name'] != name]
        clear_console()
        if(search_card):
            print(f'\n Voici la carte: {search_card["name"]}, de {search_card["owner"]} \n')
            self.cards = updatedCards
        else:
            print(f"\n La carte n'a pas été trouvée ou a été perdu \n")
    
    def add_photo(self, photoID, owner):
        new_photo = {
            "photoID": photoID,
            "owner": owner
        }
        self.photos.append(new_photo)
        print(f'\n Ajout réussi \n')

    def get_photo(self, photoID):
        search_photo = next((item for item in self.photos if item['photoID'] == photoID), None)
        updatedPhotos = [item for item in self.photos if item['photoID'] != photoID]
        clear_console()
        if(search_photo):
            print(f"\n Voici la photo : {search_photo['photoID']}, de {search_photo['owner']} \n")
            self.photos = updatedPhotos
        else:
            print(f"\n La photo n'a pas été trouvée ou a été perdu \n")